### PR TITLE
C2361 Disallow editing of Instance records that have underlying SRS MARC records (Folijet) (TaaS)

### PR DESCRIPTION
**Pull request by DuskTraum**
[C2361](https://foliotest.testrail.io/index.php?/cases/view/2361)
image
[Allure report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5585/allure/)

### CODE REVIEW

File: cypress/support/fragments/data_import/logs/fileDetails.js [hunk 1]
```diff
@@ -321,9 +321,8 @@ export default {
   },
 
   openJsonScreen: (title) => {
-    cy.get('#search-results-list').find('*[class^="mclCell"]').contains(title).focus();
     cy.get('#search-results-list')
-      .find('*[class^="mclCell"]')
+      .find('*[class^="mclCell"] a')
```
[the remark this answers is not in the document]
Comment (DuskTraum) [reply cypress/support/fragments/data_import/logs/fileDetails.js]: I think it's no needed anymore. The 'openJsonScreen' was changed already before me, I resolved conflicts within it by using their solution ( it's the same as mine actually)

File: cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const filePath = 'marcFileForC2361.mrc';
+    const marcFileName = `C2361 autotestFileName ${getRandomPostfix()}`;
+    const jobProfileToRun = 'Default - Create instance and SRS MARC Bib';
+    const title = "101 things I wish I'd known when I started";
+
+    before('login', () => {
+      cy.createTempUser([
+        Permissions.moduleDataImportEnabled.gui,
+        Permissions.settingsDataImportEnabled.gui,
+        Permissions.inventoryAll.gui,
```
Comment (TetianaParanich) [inline cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js:28]: The user should has only 'Inventory: All permissions' permission

File: cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      ]).then((userProperties) => {
+        user = userProperties;
+
+        cy.login(userProperties.username, userProperties.password, {
+          path: TopMenu.dataImportPath,
+          waiter: DataImport.waitLoading,
+        });
+
+        // upload a mrc file
```
Comment (TetianaParanich) [inline cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js:37]: please import the file as admin user before creating temp user

File: cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    it(
+      'C2361 Disallow editing of Instance records that have underlying SRS MARC records',
+      { tags: [TestTypes.extendedPath, DevTeams.folijet] },
+      () => {
+        Logs.openFileDetails(marcFileName);
+        FileDetails.verifyLogDetailsPageIsOpened();
+        FileDetails.openJsonScreen(title);
+        JsonScreenView.verifyJsonScreenIsOpened();
+        JsonScreenView.getInstanceHrid().then((hrid) => {
+          instanceHrid = hrid;
```
Comment (TetianaParanich) [inline cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js:68]: put 63-68 steps in preconditions

File: cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js (hunk 3 above)
Comment (DuskTraum) [reply cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js]: Done

File: cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js (hunk 4 above)
Comment (DuskTraum) [reply cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js]: Done

File: cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js (hunk 2 above)
Comment (DuskTraum) [reply cypress/e2e/data-import/importing-marc-bib-files/disallow-editing-of-instance-records-that-have-underlying-srs-marc-records.cy.js]: Fixed

**Review by TetianaParanich — APPROVED** [review]

**Review by dnovikau — APPROVED** [review]

**Pull request merged**